### PR TITLE
Add hex decimal to long scalar functions

### PR DESCRIPTION
**Pull request by xiangfu0**
Add hex decimal to long scalar functions

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/function/scalar/DataTypeConversionFunctions.java [hunk 1]
```diff
@@ -129,4 +129,27 @@ public static byte[] base64Encode(byte[] input) {
...
+   * Converts a hex string to the corresponded long value.
+   * @param input hex decimal string
+   * @return decoded long value
+   */
+  @ScalarFunction
+  public static long hexDecimalToLong(String input) {
+    if (input.startsWith("0x")) {
+      input = input.substring(2);
+    }
+    return Long.parseLong(input, 16);
```
Comment (gortiz) [inline pinot-common/src/main/java/org/apache/pinot/common/function/scalar/DataTypeConversionFunctions.java:143]: Tip: You can call `public static long parseLong(CharSequence s, int beginIndex, int endIndex, int radix)` instead allocating a new string here.

**Review by gortiz — APPROVED** [review]
I'm preemptively approving this, but please try to refactor the code to use the cheaper method before merging ;)

File: pinot-common/src/main/java/org/apache/pinot/common/function/scalar/DataTypeConversionFunctions.java (hunk 1 above)
Comment (xiangfu0) [reply pinot-common/src/main/java/org/apache/pinot/common/function/scalar/DataTypeConversionFunctions.java]: 👍  updated the PR

**Comment by Jackie-Jiang** [comment]
Remember to add this to the pinot doc

**Pull request merged**